### DISCUSSION TITLE
Is there a way to allow multi-line HTML to be retained?

### DISCUSSION THREAD

**Discussion by NullVoxPopuli** [remarkjs/remark#676] [Q&A] [answered]
For example, I'm seeing: 
```md
<a
  href="..."
  target="_blank"
  rel="noopener"
  data-test="my-link"
>  
  some text
</a>
```

Reproduction: https://astexplorer.net/#/gist/f2ab347d07a755a861506e32cde2c652/f8187ce6a4f853d5fb8ea4bc0b1538f4c400d259

The issue: the dangling angle bracket is interpreted as a blockquote:

![image](https://user-images.githubusercontent.com/199018/112901395-b4fd2700-90b2-11eb-9e7e-7c0d9ef92810.png)

**Comment by ChristianMurphy** [comment, marked as answer]
Remark follows the CommonMark standard, this behavior is expected https://spec.commonmark.org/dingus/?text=%3Ca%0A%20%20href%3D%22...%22%0A%20%20target%3D%22_blank%22%0A%20%20rel%3D%22noopener%22%0A%20%20data-test%3D%22my-link%22%0A%3E%0A%20%20some%20text%0A%3C%2Fa%3E%0A%0A
And is documented at https://spec.commonmark.org/0.29/#example-617

The solution is to keep the closing bracket of the open tag on the same line as the last attribute
https://astexplorer.net/#/gist/f2ab347d07a755a861506e32cde2c652/b3a2dcfa3dbfd25166afbc3214159523e2a6e3ef
documented at https://spec.commonmark.org/0.29/#example-639

**Reply by NullVoxPopuli** [reply]
is there a way to change that behavior?

parsing-wise, it seems reasonable to me to have the parser "be watching" for a `>` when a `<` is encountered

**Reply by ChristianMurphy** [reply]
Yes, if you want support in remark itself, open a thread at https://talk.commonmark.org and get the change as an approved part of CommonMark.
Or if you don't want to go through the standardization process, create a parser plugin to change the behavior of micromark/remark. See https://github.com/micromark/micromark/discussions/45 for semi-related discussion on another CommonMark customization.

That said, the easiest solution would like be to write valid HTML in Markdown, as CommonMark already defines it. :slightly_smiling_face:

**Reply by NullVoxPopuli** [reply]
a dangling `>` _is_ valid HTML, that's why I opened the issue. It's a common pattern when you have lots of attributes, and a fixed column count.

**Reply by ChristianMurphy** [reply]
> is valid HTML

I'm aware it's valid in the WhatWG HTML spec, again

> valid HTML _in Markdown, as CommonMark defines it_

CommonMark does not consider it HTML, if you disagree https://talk.commonmark.org is the place to get the specification changed

**Reply by ChristianMurphy** [reply]
The same behavior happens on GitHub by the way, as GFM is an extension of CommonMark

```markdown
<a
  href="..."
  target="_blank"
  rel="noopener"
  data-test="my-link"
>  
  some text
</a>
```

renders as:


<a
  href="..."
  target="_blank"
  rel="noopener"
  data-test="my-link"
>  
  some text
</a>

**Reply by NullVoxPopuli** [reply]
Gonna gather feedback before I go and PR the parser: https://talk.commonmark.org/t/thoughts-on-changing-supported-html-syntax-to-include-the-dangling-closing-angle-bracket/3827
